### ISSUE TITLE
feat : Sub-Tasks

### ISSUE THREAD

**Issue by tu2-atmanand** [tu2-atmanand/Task-Board#12]
I want to implement the Sub-Task functionality in two phases : 

## First phase
> Need to implement this in the first release v1.0.0

For this first phase of implementing the subtasks, the sub-tasks will be simply like the body of the main task. So for this, my json structure will contain an extra field called as title, which will act like the main, compulsory field, to store the main task or title of the task. And the body will now be replaced by the body content for this task, where the user can either put more content regarding this task or the subtasks will be added in this field. So the idea is to design by taking inspiration from the Trello UI/UX : 

![image](https://github.com/user-attachments/assets/d87a2ab9-23c1-4a2a-9f56-fc22c4971929)

So, as seen from the above image, Description can be changed as Task Body, in which the user will be able to add any information or even images in the future features. Before the Task Body, provide another section to add Sub-Task, so after the user enters the sub-task and hit enter, it will be added as a sub checkbox task. 

To add the subtasks, in the body, the body will be an string array field. The tasks will be stored along the with pattern `- [ ]` or `-'x]`, based on which the status of the subtasks can be known. If the element of the body array doesnt start with this types of patterns, then that element will simply be the body of the Main task.

So to summarize, in the above issue following taks are remaining : 
- [x] Improve the UI and Layout of the EditTaskModal and AddTaskModal as per the Trello UI.
- [x] Create feature to fold unfold the Body/Description of the task when it is displayed in the TaskItem Card on the TaskBoard.
- [x] A delete icon on right of each subtasks to delete the empty or new added subtask in the EditTaskModal. So once the user deletes the subtas, that element needs to be removed from the body array.

> I think it will be a good idea to keep the Sub-Tasks visible in the board, if in future, i got good number of feedback, i can provide an option to fold/unfold the Sub-tasks.

## Second Phase
Find the second phase of implementing the subtasks or improving them in this issue : #13

**Comment by tu2-atmanand** [comment]
So to summarize, in the above issue following taks are remaining : 
- [ ] A new option in setting to allow user, if the user wants to see the body of the task in the TaskBoard or it should be hidden.
- [ ] Improve the UI and Layout of the EditTaskModal and AddTaskModal as per the Trello UI.
- [ ] Feature to fold/unfold the subtask in the main taskItem.

**Comment by tu2-atmanand** [comment]
Adding subtasks using AddTaskModal and Detecting and extracting subtasks from ScanningVaults.ts file code has been implemented and working properly.

- [ ] Edit the code of EditTaskModal to work with the subtasks.
